### PR TITLE
Dark Theme setup and button theming

### PR DESCRIPTION
**Pull request by GuillaumeRx**
Explanation: Add `@metamask/design-tokens` that includes dark theme color variables and apply it on the button component

### CODE REVIEW

**Comment by GuillaumeRx** [comment]
Waiting on https://github.com/MetaMask/design-tokens/pull/22 to be merged and deployed to continue on button theming

**Comment by brad-decker** [comment]
omg i love this so much.

**Review by brad-decker — DISMISSED** [review]

**Comment by darkwing** [comment]
Seeing this on the home screen:

https://user-images.githubusercontent.com/46655/155357268-d0aab679-bb17-475b-9c3d-f046ccf7af03.mp4

**Comment by GuillaumeRx** [comment]
> Seeing this on the home screen:
> 
>  Hover.mp4

@darkwing oh yes good catch, we don't have a darker in between that matches the original hover color. @georgewrmarshall what's you thoughts about this ?

**Comment by darkwing** [comment]
This issue is also happening on the "Settings" > "About" screen on `btn-link`

File: .storybook/preview.js [hunk 1]
```diff
@@ -72,8 +73,30 @@ const proxiedBackground = new Proxy(
...
+      : document.documentElement.classList.remove('theme-dark');
+
+    // @TODO: change when data-theme is implemented in design-tokens
+    //const currentTheme = document.documentElement.getAttribute('data-theme');
+    // if (currentTheme === 'light' && isDark) {
+    //   document.documentElement.setAttribute('data-theme', 'dark');
+    // } else if (currentTheme === 'dark' && !isDark) {
+    //   document.documentElement.setAttribute('data-theme', 'light');
+    // }
+  }, [isDark]);
```
Comment (georgewrmarshall) [inline .storybook/preview.js:98]: V`1.3.0` has been release so this can be implemented.

**Review by georgewrmarshall — COMMENTED** [review]
Looking good. I've added some of my suggestions to a branch that I've created a PR into this branch for https://github.com/MetaMask/metamask-extension/pull/13742

**Review by georgewrmarshall — DISMISSED** [review]
Looks great! Love the dark theme / background implementation in storybook 💯 I'll update the `Box` PR to revert to develop so this implementation can be used.

**Review by darkwing — DISMISSED** [review]

**Review by darkwing — APPROVED** [review]

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Pull request merged**